src/nvim/buffer.c

```diff
@@ -5329,6 +5332,7 @@ void bufhl_clear_line_range(buf_T *buf,
 {
   linenr_T first_changed = MAXLNUM, last_changed = -1;
 
+  // TODO(bfredl): implement kb_itr_interval to jump directly to the first line
```

Comment:
I think this comment is obsolete: the `kb_itr_get`/`kb_iter_next` dance should already do this in most efficient manner.